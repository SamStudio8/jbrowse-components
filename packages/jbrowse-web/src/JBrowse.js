import '@gmod/jbrowse-core/fonts/material-icons.css'
import CircularProgress from '@material-ui/core/CircularProgress'
import CssBaseline from '@material-ui/core/CssBaseline'
import { ThemeProvider } from '@material-ui/styles'
import { observer } from 'mobx-react'
import React from 'react'
import 'typeface-roboto'
import App from './ui/App'
import Theme from './ui/theme'

export default observer(({ state }) => {
  let DisplayComponent = (
    <CircularProgress
      style={{
        position: 'fixed',
        top: '50%',
        left: '50%',
        marginTop: -250,
        marginLeft: -250,
      }}
      size={500}
    />
  )
  if (state.errorMessage) DisplayComponent = <div>{state.errorMessage}</div>
  if (state.activeSession)
    DisplayComponent = (
      <App
        session={state.activeSession}
        sessionNames={state.sessionNames}
        setActiveSession={state.activateSession}
        addSession={state.addSession}
      />
    )

  return (
    <ThemeProvider theme={Theme}>
      <CssBaseline />
      {DisplayComponent}
    </ThemeProvider>
  )
})
