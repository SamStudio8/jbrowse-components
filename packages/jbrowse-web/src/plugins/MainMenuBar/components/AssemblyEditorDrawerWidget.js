import Avatar from '@material-ui/core/Avatar'
import Button from '@material-ui/core/Button'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import CardHeader from '@material-ui/core/CardHeader'
import ClickAwayListener from '@material-ui/core/ClickAwayListener'
import Divider from '@material-ui/core/Divider'
import Fab from '@material-ui/core/Fab'
import Grow from '@material-ui/core/Grow'
import Icon from '@material-ui/core/Icon'
import IconButton from '@material-ui/core/IconButton'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemAvatar from '@material-ui/core/ListItemIcon'
import ListItemText from '@material-ui/core/ListItemText'
import ListSubheader from '@material-ui/core/ListSubheader'
import Paper from '@material-ui/core/Paper'
import Popper from '@material-ui/core/Popper'
import TextField from '@material-ui/core/TextField'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'
import { PropTypes as MobxPropTypes } from 'mobx-react'
import { observer } from 'mobx-react-lite'
import { getRoot, getSnapshot } from 'mobx-state-tree'
import propTypes from 'prop-types'
import React, { useState, useEffect } from 'react'
import { readConfObject } from '@gmod/jbrowse-core/configuration'

const styles = theme => ({
  root: {
    textAlign: 'left',
    padding: theme.spacing.unit,
  },
  header: {
    display: 'flex',
    alignItems: 'center',
  },
  headerText: {
    flex: 'auto',
  },
  paper: {
    marginBottom: theme.spacing.unit,
  },
  popper: {
    zIndex: theme.zIndex.drawer + 50,
  },
  assembly: {
    marginBottom: theme.spacing.unit,
  },
  listNumber: {
    width: 24,
    height: 24,
    fontSize: '80%',
  },
  fab: {
    float: 'right',
    position: 'sticky',
    'margin-top': theme.spacing.unit * 2,
    bottom: theme.spacing.unit * 2,
    right: theme.spacing.unit * 2,
  },
  menu: {
    padding: theme.spacing.unit * 2,
    maxWidth: 400,
  },
  menuButtons: {
    display: 'flex',
    justifyContent: 'flex-end',
  },
})

function AssemblyEditorDrawerWidget(props) {
  const [addMenuAnchorEl, setAddMenuAnchorEl] = useState(null)
  const [helpMenuAnchorEl, setHelpMenuAnchorEl] = useState(null)
  const [newAssemblyName, setNewAssemblyName] = useState('')
  const [assemblies, setAssemblies] = useState(new Map())

  const { classes, model } = props
  const rootModel = getRoot(model)

  function handleHelpClick(event) {
    setHelpMenuAnchorEl(helpMenuAnchorEl ? null : event.currentTarget)
  }

  function handleHelpClose(event) {
    if (helpMenuAnchorEl && helpMenuAnchorEl.contains(event.target)) return
    setHelpMenuAnchorEl(null)
  }

  function handleFabClick(event) {
    setAddMenuAnchorEl(addMenuAnchorEl ? null : event.currentTarget)
  }

  function handleFabClose(event) {
    if (addMenuAnchorEl && addMenuAnchorEl.contains(event.target)) return
    setAddMenuAnchorEl(null)
  }

  function handleNewAssemblyChange(event) {
    setNewAssemblyName(event.target.value)
  }

  function handleAddAssembly(event) {
    setAssemblies(new Map())
    handleFabClose(event)
    rootModel.configuration.addAssembly(newAssemblyName)
    setNewAssemblyName('')
  }

  async function fetchAssemblies(rootAssemblies) {
    const results = []
    for (const [assemblyName, assembly] of rootAssemblies) {
      const { rpcManager } = rootModel
      results.push(
        rpcManager.call(
          assembly.configId,
          'getRefNameAliases',
          {
            sessionId: assemblyName,
            adapterType: readConfObject(assembly, [
              'refNameAliases',
              'adapter',
              'type',
            ]),
            adapterConfig: readConfObject(assembly, [
              'refNameAliases',
              'adapter',
            ]),
          },
          { timeout: 1000000 },
        ),
      )
    }
    const assemblyRefNameAliases = await Promise.all(results)
    const newAssemblies = new Map()
    let idx = 0
    for (const [assemblyName, assembly] of rootAssemblies) {
      newAssemblies.set(assemblyName, [assembly, assemblyRefNameAliases[idx]])
      idx += 1
    }
    setAssemblies(newAssemblies)
  }

  useEffect(
    () => {
      fetchAssemblies(rootModel.configuration.assemblies)
    },
    [getSnapshot(rootModel.configuration.assemblies)],
  )

  return (
    <div className={classes.root}>
      <div className={classes.header}>
        <Typography variant="subtitle1" className={classes.headerText}>
          Configured assemblies
        </Typography>
        <IconButton
          onClick={handleHelpClick}
          data-testid="AssemblyEditorDrawerWidgetHelpButton"
        >
          <Icon>help</Icon>
        </IconButton>
        <Popper
          className={classes.popper}
          anchorEl={helpMenuAnchorEl}
          open={Boolean(helpMenuAnchorEl)}
          transition
          data-testid="AssemblyEditorDrawerWidgetHelpPopper"
        >
          {({ TransitionProps }) => (
            <Grow {...TransitionProps} in={Boolean(helpMenuAnchorEl)}>
              <Paper className={classes.menu}>
                <ClickAwayListener onClickAway={handleHelpClose}>
                  <div>
                    <Typography paragraph>
                      JBrowse uses the assemblies listed here to help make sure
                      that tracks appear on the correct reference sequence.
                    </Typography>
                    <Typography paragraph>
                      You can define aliases for an assembly so that the two
                      assembly names are treated as the same, e.g. GRCh37 and
                      hg19.
                    </Typography>
                    <Typography paragraph>
                      You can also define any reference sequences within that
                      assembly that may have multiple names, such as 1 and chr1
                      or ctgA and contigA.
                    </Typography>
                  </div>
                </ClickAwayListener>
              </Paper>
            </Grow>
          )}
        </Popper>
      </div>
      {Array.from(assemblies, ([assemblyName, [assembly, refNameAliases]]) => (
        <Card className={classes.assembly} key={assemblyName}>
          <CardHeader
            title={assemblyName}
            action={
              <>
                <IconButton
                  onClick={() =>
                    rootModel.editConfiguration(
                      rootModel.configuration.assemblies.get(assemblyName),
                    )
                  }
                >
                  <Icon>edit</Icon>
                </IconButton>
                <IconButton
                  onClick={() =>
                    rootModel.configuration.removeAssembly(assemblyName)
                  }
                >
                  <Icon>delete</Icon>
                </IconButton>
              </>
            }
          />
          <CardContent>
            <Paper className={classes.paper}>
              <List dense subheader={<ListSubheader>Aliases:</ListSubheader>}>
                <Divider />
                {readConfObject(assembly, 'aliases').map((alias, idx) => (
                  <ListItem key={alias}>
                    <ListItemAvatar>
                      <Avatar className={classes.listNumber}>{idx + 1}</Avatar>
                    </ListItemAvatar>
                    <ListItemText>{alias}</ListItemText>
                  </ListItem>
                ))}
              </List>
            </Paper>
            <Paper>
              <List
                dense
                subheader={
                  <ListSubheader>Reference Sequence Aliases:</ListSubheader>
                }
              >
                <Divider />
                {refNameAliases.map(({ refName, aliases }) => (
                  <List
                    key={refName}
                    dense
                    subheader={<ListSubheader>{refName}</ListSubheader>}
                  >
                    {aliases.map((alias, idx) => (
                      <ListItem key={alias}>
                        <ListItemAvatar>
                          <Avatar className={classes.listNumber}>
                            {idx + 1}
                          </Avatar>
                        </ListItemAvatar>
                        <ListItemText>{alias}</ListItemText>
                      </ListItem>
                    ))}
                    <Divider />
                  </List>
                ))}
              </List>
            </Paper>
          </CardContent>
        </Card>
      ))}
      <Fab
        color="secondary"
        className={classes.fab}
        onClick={handleFabClick}
        data-testid="AssemblyEditorDrawerWidgetFAB"
      >
        <Icon>add</Icon>
      </Fab>
      <Popper
        className={classes.popper}
        anchorEl={addMenuAnchorEl}
        open={Boolean(addMenuAnchorEl)}
        transition
        data-testid="AssemblyEditorDrawerWidgetAddPopper"
      >
        {({ TransitionProps }) => (
          <Grow {...TransitionProps} in={Boolean(addMenuAnchorEl)}>
            <Paper className={classes.menu}>
              <ClickAwayListener onClickAway={handleFabClose}>
                <div>
                  <Typography
                    component="p"
                    variant="subtitle1"
                    color="textSecondary"
                  >
                    Enter the name of the new assembly
                  </Typography>
                  <TextField
                    autoFocus
                    label="Assembly name"
                    value={newAssemblyName}
                    onChange={handleNewAssemblyChange}
                  />
                  <div className={classes.menuButtons}>
                    <Button color="secondary" onClick={handleFabClose}>
                      Cancel
                    </Button>
                    <Button
                      color="secondary"
                      onClick={handleAddAssembly}
                      disabled={!newAssemblyName}
                      data-testid="AssemblyEditorDrawerWidgetSubmitButton"
                    >
                      Add assembly
                    </Button>
                  </div>
                </div>
              </ClickAwayListener>
            </Paper>
          </Grow>
        )}
      </Popper>
    </div>
  )
}

AssemblyEditorDrawerWidget.propTypes = {
  model: MobxPropTypes.observableObject.isRequired,
  classes: propTypes.objectOf(propTypes.string).isRequired,
}

export default withStyles(styles)(observer(AssemblyEditorDrawerWidget))
