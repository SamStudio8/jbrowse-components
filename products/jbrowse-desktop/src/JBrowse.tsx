import React, { useEffect, useState, Suspense } from 'react'
import { observer } from 'mobx-react'
import { CssBaseline, ThemeProvider } from '@material-ui/core'
import { getConf } from '@jbrowse/core/configuration'
import { App, createJBrowseTheme } from '@jbrowse/core/ui'
import PluginManager from '@jbrowse/core/PluginManager'
import { AssemblyManager } from '@jbrowse/plugin-data-management'
import deepmerge from 'deepmerge'

// locals
import { RootModel } from './rootModel'

const JBrowse = observer(
  ({ pluginManager, mode }: { mode: string; pluginManager: PluginManager }) => {
    const { rootModel } = pluginManager

    return rootModel ? (
      <JBrowseNonNullRoot mode={mode} rootModel={rootModel as RootModel} />
    ) : null
  },
)

const JBrowseNonNullRoot = observer(
  ({ rootModel, mode }: { mode: string; rootModel: RootModel }) => {
    const [firstLoad, setFirstLoad] = useState(true)
    const { session, jbrowse, error, isAssemblyEditing, setAssemblyEditing } =
      rootModel

    useEffect(() => {
      if (firstLoad && session) {
        setFirstLoad(false)
      }
    }, [firstLoad, session])

    if (error) {
      throw error
    }

    const theme = getConf(jbrowse, 'theme')

    const colorTheme = React.useMemo(
      () => ({
        palette: {
          type: mode,
        },
      }),
      [mode],
    )

    return (
      // @ts-ignore
      <ThemeProvider theme={createJBrowseTheme(deepmerge(theme, colorTheme))}>
        <CssBaseline />
        {session ? (
          <>
            <App session={session} />
            <Suspense fallback={<div />}>
              {isAssemblyEditing ? (
                <AssemblyManager
                  rootModel={rootModel}
                  onClose={() => {
                    setAssemblyEditing(false)
                  }}
                />
              ) : null}
            </Suspense>
          </>
        ) : (
          <div>No session</div>
        )}
      </ThemeProvider>
    )
  },
)

export default JBrowse
