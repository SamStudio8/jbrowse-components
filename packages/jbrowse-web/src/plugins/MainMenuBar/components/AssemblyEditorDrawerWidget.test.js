import { render, cleanup, fireEvent, wait } from 'react-testing-library'
import React from 'react'
import { createTestEnv } from '../../../JBrowse'
import AssemblyEditorDrawerWidget from './AssemblyEditorDrawerWidget'

jest.mock('popper.js', () => {
  const PopperJS = jest.requireActual('popper.js')
  return class Popper {
    static placements = PopperJS.placements

    constructor() {
      return {
        destroy: () => {},
        scheduleUpdate: () => {},
      }
    }
  }
})

describe('<AssemblyEditorDrawerWidget />', () => {
  let model

  beforeAll(async () => {
    const { rootModel } = await createTestEnv({
      configId: 'testing',
      assemblies: {
        volvox: {
          configId: 'volvox',
          aliases: ['vvx'],
          refNameAliases: {
            adapter: {
              type: 'FromConfigAdapter',
              refNameAliases: [
                {
                  refName: 'ctgA',
                  aliases: ['A', 'contigA'],
                },
                {
                  refName: 'ctgB',
                  aliases: ['B', 'contigB'],
                },
              ],
            },
          },
          sequence: {
            configId: 'iTo6LoXUeJ',
            type: 'ReferenceSequence',
            adapter: {
              configId: 'Zd0NLmtxPZ3',
              type: 'FromConfigAdapter',
              regions: [
                {
                  refName: 'fakeContig',
                  start: 0,
                  end: 1000,
                },
              ],
            },
          },
        },
      },
    })
    rootModel.addDrawerWidget(
      'HierarchicalTrackSelectorDrawerWidget',
      'hierarchicalTrackSelectorDrawerWidget',
    )
    model = rootModel.drawerWidgets.get('hierarchicalTrackSelectorDrawerWidget')
  })

  afterEach(cleanup)

  it('renders', async () => {
    const { container, getByText } = render(
      <AssemblyEditorDrawerWidget model={model} />,
    )
    await wait(() => getByText('volvox'))
    expect(container.firstChild).toMatchSnapshot()
  })

  it('opens and closes dialog on FAB click', async () => {
    const { getByTestId, getByText } = render(
      <AssemblyEditorDrawerWidget model={model} />,
    )
    await wait(() => getByText('volvox'))
    expect(() => getByText(/Enter the name/)).toThrow()
    fireEvent.click(getByTestId('AssemblyEditorDrawerWidgetFAB'))
    expect(getByTestId('AssemblyEditorDrawerWidgetAddPopper')).toMatchSnapshot()
    fireEvent.click(getByTestId('AssemblyEditorDrawerWidgetFAB'))
    expect(getByTestId('AssemblyEditorDrawerWidgetAddPopper')).toMatchSnapshot()
  })

  it('opens and closes dialog on help click', async () => {
    const { getByTestId, getByText } = render(
      <AssemblyEditorDrawerWidget model={model} />,
    )
    await wait(() => getByText('volvox'))
    expect(() => getByText(/JBrowse uses the assemblies/)).toThrow()
    fireEvent.click(getByTestId('AssemblyEditorDrawerWidgetHelpButton'))
    expect(
      getByTestId('AssemblyEditorDrawerWidgetHelpPopper'),
    ).toMatchSnapshot()
    fireEvent.click(getByTestId('AssemblyEditorDrawerWidgetHelpButton'))
    expect(
      getByTestId('AssemblyEditorDrawerWidgetHelpPopper'),
    ).toMatchSnapshot()
  })

  it('adds an assembly', async () => {
    const { getByTestId, getByText, getByDisplayValue } = render(
      <AssemblyEditorDrawerWidget model={model} />,
    )
    await wait(() => getByText('volvox'))
    fireEvent.click(getByTestId('AssemblyEditorDrawerWidgetFAB'))
    fireEvent.change(getByDisplayValue(''), {
      target: { value: 'testAssembly' },
    })
    fireEvent.click(getByTestId('AssemblyEditorDrawerWidgetSubmitButton'))
    await wait(() => getByText('testAssembly'))

    // expect(getByTestId('AssemblyEditorDrawerWidgetAddPopper')).toMatchSnapshot()
    // fireEvent.click(getByTestId('AssemblyEditorDrawerWidgetFAB'))
    // expect(getByTestId('AssemblyEditorDrawerWidgetAddPopper')).toMatchSnapshot()

    // const wrapper = shallow(<AssemblyEditorDrawerWidget model={model} />)
    // const instance = wrapper.instance()
    // instance.handleNewAssemblyChange({ target: { value: 'abc' } })
    // expect(wrapper.state('newAssemblyName')).toBe('abc')
    // instance.handleAddAssembly({ target: 'theAddAssemblyButton' })
    // expect(wrapper).toMatchSnapshot()
  })
})
