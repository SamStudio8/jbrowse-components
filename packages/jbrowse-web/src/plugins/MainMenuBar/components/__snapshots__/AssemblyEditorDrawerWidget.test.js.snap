// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AssemblyEditorDrawerWidget /> opens and closes dialog on FAB click 1`] = `
<div
  class="AssemblyEditorDrawerWidget-popper-158"
  data-testid="AssemblyEditorDrawerWidgetAddPopper"
  role="tooltip"
  style="position: absolute;"
>
  <div
    class="MuiPaper-root-236 MuiPaper-elevation2-240 MuiPaper-rounded-237 AssemblyEditorDrawerWidget-menu-162"
    style="opacity: 1; transform: scale(1, 1);"
  >
    <div>
      <p
        class="MuiTypography-root-164 MuiTypography-subtitle1-182 MuiTypography-colorTextSecondary-197"
      >
        Enter the name of the new assembly
      </p>
      <div
        class="MuiFormControl-root-307"
      >
        <label
          class="MuiFormLabel-root-322 MuiFormLabel-focused-323 MuiInputLabel-focused-312 MuiInputLabel-root-311 MuiInputLabel-formControl-316 MuiInputLabel-animated-319 MuiInputLabel-shrink-318"
          data-shrink="true"
        >
          Assembly name
        </label>
        <div
          class="MuiInputBase-root-342 MuiInput-root-329 MuiInput-underline-333 MuiInputBase-focused-344 MuiInput-focused-331 MuiInputBase-formControl-343 MuiInput-formControl-330"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input-352 MuiInput-input-337"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="AssemblyEditorDrawerWidget-menuButtons-163"
      >
        <button
          class="MuiButtonBase-root-206 MuiButton-root-359 MuiButton-text-361 MuiButton-textSecondary-363 MuiButton-flat-364 MuiButton-flatSecondary-366"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-360"
          >
            Cancel
          </span>
          <span
            class="MuiTouchRipple-root-228"
          />
        </button>
        <button
          class="MuiButtonBase-root-206 MuiButtonBase-disabled-207 MuiButton-root-359 MuiButton-text-361 MuiButton-textSecondary-363 MuiButton-flat-364 MuiButton-flatSecondary-366 MuiButton-disabled-379"
          data-testid="AssemblyEditorDrawerWidgetSubmitButton"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label-360"
          >
            Add assembly
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AssemblyEditorDrawerWidget /> opens and closes dialog on FAB click 2`] = `
<div
  class="AssemblyEditorDrawerWidget-popper-158"
  data-testid="AssemblyEditorDrawerWidgetAddPopper"
  role="tooltip"
  style="position: absolute;"
>
  <div
    class="MuiPaper-root-236 MuiPaper-elevation2-240 MuiPaper-rounded-237 AssemblyEditorDrawerWidget-menu-162"
    style="opacity: 0; transform: scale(0.75, 0.5625);"
  >
    <div>
      <p
        class="MuiTypography-root-164 MuiTypography-subtitle1-182 MuiTypography-colorTextSecondary-197"
      >
        Enter the name of the new assembly
      </p>
      <div
        class="MuiFormControl-root-307"
      >
        <label
          class="MuiFormLabel-root-322 MuiFormLabel-focused-323 MuiInputLabel-focused-312 MuiInputLabel-root-311 MuiInputLabel-formControl-316 MuiInputLabel-animated-319 MuiInputLabel-shrink-318"
          data-shrink="true"
        >
          Assembly name
        </label>
        <div
          class="MuiInputBase-root-342 MuiInput-root-329 MuiInput-underline-333 MuiInputBase-focused-344 MuiInput-focused-331 MuiInputBase-formControl-343 MuiInput-formControl-330"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input-352 MuiInput-input-337"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="AssemblyEditorDrawerWidget-menuButtons-163"
      >
        <button
          class="MuiButtonBase-root-206 MuiButton-root-359 MuiButton-text-361 MuiButton-textSecondary-363 MuiButton-flat-364 MuiButton-flatSecondary-366"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-360"
          >
            Cancel
          </span>
          <span
            class="MuiTouchRipple-root-228"
          />
        </button>
        <button
          class="MuiButtonBase-root-206 MuiButtonBase-disabled-207 MuiButton-root-359 MuiButton-text-361 MuiButton-textSecondary-363 MuiButton-flat-364 MuiButton-flatSecondary-366 MuiButton-disabled-379"
          data-testid="AssemblyEditorDrawerWidgetSubmitButton"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label-360"
          >
            Add assembly
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AssemblyEditorDrawerWidget /> opens and closes dialog on help click 1`] = `
<div
  class="AssemblyEditorDrawerWidget-popper-389"
  data-testid="AssemblyEditorDrawerWidgetHelpPopper"
  role="tooltip"
  style="position: absolute;"
>
  <div
    class="MuiPaper-root-467 MuiPaper-elevation2-471 MuiPaper-rounded-468 AssemblyEditorDrawerWidget-menu-393"
    style="opacity: 1; transform: scale(1, 1);"
  >
    <div>
      <p
        class="MuiTypography-root-395 MuiTypography-body1-404 MuiTypography-paragraph-423"
      >
        JBrowse uses the assemblies listed here to help make sure that tracks appear on the correct reference sequence.
      </p>
      <p
        class="MuiTypography-root-395 MuiTypography-body1-404 MuiTypography-paragraph-423"
      >
        You can define aliases for an assembly so that the two assembly names are treated as the same, e.g. GRCh37 and hg19.
      </p>
      <p
        class="MuiTypography-root-395 MuiTypography-body1-404 MuiTypography-paragraph-423"
      >
        You can also define any reference sequences within that assembly that may have multiple names, such as 1 and chr1 or ctgA and contigA.
      </p>
    </div>
  </div>
</div>
`;

exports[`<AssemblyEditorDrawerWidget /> opens and closes dialog on help click 2`] = `
<div
  class="AssemblyEditorDrawerWidget-popper-389"
  data-testid="AssemblyEditorDrawerWidgetHelpPopper"
  role="tooltip"
  style="position: absolute;"
>
  <div
    class="MuiPaper-root-467 MuiPaper-elevation2-471 MuiPaper-rounded-468 AssemblyEditorDrawerWidget-menu-393"
    style="opacity: 0; transform: scale(0.75, 0.5625);"
  >
    <div>
      <p
        class="MuiTypography-root-395 MuiTypography-body1-404 MuiTypography-paragraph-423"
      >
        JBrowse uses the assemblies listed here to help make sure that tracks appear on the correct reference sequence.
      </p>
      <p
        class="MuiTypography-root-395 MuiTypography-body1-404 MuiTypography-paragraph-423"
      >
        You can define aliases for an assembly so that the two assembly names are treated as the same, e.g. GRCh37 and hg19.
      </p>
      <p
        class="MuiTypography-root-395 MuiTypography-body1-404 MuiTypography-paragraph-423"
      >
        You can also define any reference sequences within that assembly that may have multiple names, such as 1 and chr1 or ctgA and contigA.
      </p>
    </div>
  </div>
</div>
`;

exports[`<AssemblyEditorDrawerWidget /> renders 1`] = `
<div
  class="AssemblyEditorDrawerWidget-root-1"
>
  <div
    class="AssemblyEditorDrawerWidget-header-2"
  >
    <h6
      class="MuiTypography-root-11 MuiTypography-subtitle1-29 AssemblyEditorDrawerWidget-headerText-3"
    >
      Configured assemblies
    </h6>
    <button
      class="MuiButtonBase-root-53 MuiIconButton-root-47"
      data-testid="AssemblyEditorDrawerWidgetHelpButton"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label-52"
      >
        <span
          aria-hidden="true"
          class="material-icons MuiIcon-root-56"
        >
          help
        </span>
      </span>
      <span
        class="MuiTouchRipple-root-75"
      />
    </button>
  </div>
  <div
    class="MuiPaper-root-83 MuiPaper-elevation1-86 MuiPaper-rounded-84 MuiCard-root-82 AssemblyEditorDrawerWidget-assembly-6"
  >
    <div
      class="MuiCardHeader-root-110"
    >
      <div
        class="MuiCardHeader-content-113"
      >
        <span
          class="MuiTypography-root-11 MuiTypography-headline-16 MuiCardHeader-title-114"
        >
          volvox
        </span>
      </div>
      <div
        class="MuiCardHeader-action-112"
      >
        <button
          class="MuiButtonBase-root-53 MuiIconButton-root-47"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-52"
          >
            <span
              aria-hidden="true"
              class="material-icons MuiIcon-root-56"
            >
              edit
            </span>
          </span>
          <span
            class="MuiTouchRipple-root-75"
          />
        </button>
        <button
          class="MuiButtonBase-root-53 MuiIconButton-root-47"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label-52"
          >
            <span
              aria-hidden="true"
              class="material-icons MuiIcon-root-56"
            >
              delete
            </span>
          </span>
          <span
            class="MuiTouchRipple-root-75"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiCardContent-root-116"
    >
      <div
        class="MuiPaper-root-83 MuiPaper-elevation2-87 MuiPaper-rounded-84 AssemblyEditorDrawerWidget-paper-4"
      >
        <ul
          class="MuiList-root-117 MuiList-dense-119 MuiList-padding-118 MuiList-subheader-120"
        >
          <li
            class="MuiListSubheader-root-121 MuiListSubheader-sticky-126 MuiListSubheader-gutters-124"
          >
            Aliases:
          </li>
          <hr
            class="MuiDivider-root-127"
          />
          <li
            class="MuiListItem-root-132 MuiListItem-default-135 MuiListItem-dense-136 MuiListItem-gutters-140"
          >
            <div
              class="MuiListItemIcon-root-144"
            >
              <div
                class="MuiAvatar-root-145 MuiAvatar-colorDefault-146 AssemblyEditorDrawerWidget-listNumber-7"
              >
                1
              </div>
            </div>
            <div
              class="MuiListItemText-root-148 MuiListItemText-dense-150"
            >
              <span
                class="MuiTypography-root-11 MuiTypography-subheading-18 MuiListItemText-primary-151 MuiListItemText-textDense-153"
              >
                vvx
              </span>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="MuiPaper-root-83 MuiPaper-elevation2-87 MuiPaper-rounded-84"
      >
        <ul
          class="MuiList-root-117 MuiList-dense-119 MuiList-padding-118 MuiList-subheader-120"
        >
          <li
            class="MuiListSubheader-root-121 MuiListSubheader-sticky-126 MuiListSubheader-gutters-124"
          >
            Reference Sequence Aliases:
          </li>
          <hr
            class="MuiDivider-root-127"
          />
          <ul
            class="MuiList-root-117 MuiList-dense-119 MuiList-padding-118 MuiList-subheader-120"
          >
            <li
              class="MuiListSubheader-root-121 MuiListSubheader-sticky-126 MuiListSubheader-gutters-124"
            >
              ctgA
            </li>
            <li
              class="MuiListItem-root-132 MuiListItem-default-135 MuiListItem-dense-136 MuiListItem-gutters-140"
            >
              <div
                class="MuiListItemIcon-root-144"
              >
                <div
                  class="MuiAvatar-root-145 MuiAvatar-colorDefault-146 AssemblyEditorDrawerWidget-listNumber-7"
                >
                  1
                </div>
              </div>
              <div
                class="MuiListItemText-root-148 MuiListItemText-dense-150"
              >
                <span
                  class="MuiTypography-root-11 MuiTypography-subheading-18 MuiListItemText-primary-151 MuiListItemText-textDense-153"
                >
                  A
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root-132 MuiListItem-default-135 MuiListItem-dense-136 MuiListItem-gutters-140"
            >
              <div
                class="MuiListItemIcon-root-144"
              >
                <div
                  class="MuiAvatar-root-145 MuiAvatar-colorDefault-146 AssemblyEditorDrawerWidget-listNumber-7"
                >
                  2
                </div>
              </div>
              <div
                class="MuiListItemText-root-148 MuiListItemText-dense-150"
              >
                <span
                  class="MuiTypography-root-11 MuiTypography-subheading-18 MuiListItemText-primary-151 MuiListItemText-textDense-153"
                >
                  contigA
                </span>
              </div>
            </li>
            <hr
              class="MuiDivider-root-127"
            />
          </ul>
          <ul
            class="MuiList-root-117 MuiList-dense-119 MuiList-padding-118 MuiList-subheader-120"
          >
            <li
              class="MuiListSubheader-root-121 MuiListSubheader-sticky-126 MuiListSubheader-gutters-124"
            >
              ctgB
            </li>
            <li
              class="MuiListItem-root-132 MuiListItem-default-135 MuiListItem-dense-136 MuiListItem-gutters-140"
            >
              <div
                class="MuiListItemIcon-root-144"
              >
                <div
                  class="MuiAvatar-root-145 MuiAvatar-colorDefault-146 AssemblyEditorDrawerWidget-listNumber-7"
                >
                  1
                </div>
              </div>
              <div
                class="MuiListItemText-root-148 MuiListItemText-dense-150"
              >
                <span
                  class="MuiTypography-root-11 MuiTypography-subheading-18 MuiListItemText-primary-151 MuiListItemText-textDense-153"
                >
                  B
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root-132 MuiListItem-default-135 MuiListItem-dense-136 MuiListItem-gutters-140"
            >
              <div
                class="MuiListItemIcon-root-144"
              >
                <div
                  class="MuiAvatar-root-145 MuiAvatar-colorDefault-146 AssemblyEditorDrawerWidget-listNumber-7"
                >
                  2
                </div>
              </div>
              <div
                class="MuiListItemText-root-148 MuiListItemText-dense-150"
              >
                <span
                  class="MuiTypography-root-11 MuiTypography-subheading-18 MuiListItemText-primary-151 MuiListItemText-textDense-153"
                >
                  contigB
                </span>
              </div>
            </li>
            <hr
              class="MuiDivider-root-127"
            />
          </ul>
        </ul>
      </div>
    </div>
  </div>
  <button
    class="MuiButtonBase-root-53 MuiFab-root-65 MuiFab-secondary-68 AssemblyEditorDrawerWidget-fab-8"
    data-testid="AssemblyEditorDrawerWidgetFAB"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiFab-label-66"
    >
      <span
        aria-hidden="true"
        class="material-icons MuiIcon-root-56"
      >
        add
      </span>
    </span>
    <span
      class="MuiTouchRipple-root-75"
    />
  </button>
</div>
`;
